File: solution/1_FindMinValueInList/1_FindMinValueInList/FindMinValueInListTest.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Linq;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using listutils;

namespace test
{
    [TestClass]
    public class FindMinValueInListTest
    {
        [TestMethod]
        [ExpectedException(typeof(ArgumentException))]
	    public void emptyListThrowsIllegalArgumentException(){
		    ListUtil.findMinValueInList();
	    }

        [TestMethod]
	    public void geefkleinste() {
		    Assert.AreEqual(1, ListUtil.findMinValueInList(4, 2, 1, 5, 3, 1, 5, 9));
	    }
    }
}

File: workspace/1_FindMinValueInList/test/listutils/FindMinValueInListTest.java
package listutils;

import static junit.framework.Assert.assertEquals;

import org.junit.Test;


public class FindMinValueInListTest {
	
	@Test(expected=IllegalArgumentException.class)
	public void emptyListThrowsIllegalArgumentException(){
		ListUtil.findMinValueInList();
	}
	
	@Test
	public void returnSmallest() {
		assertEquals(new Integer(1), ListUtil.findMinValueInList(4, 2, 1, 5, 3, 1, 5, 9));
	}

}
